### PR TITLE
Generate test values in json, to be consumed by unit tests

### PR DESCRIPTION
**Pull request by theplasmastorm**

### CODE REVIEW

File: Services.Test/TestConstants.Properties.dbl [hunk 1]
```diff
@@ -12,110 +12,139 @@
...
+        private static instance, @TestConstants, ^null
+        public static property Instance, @TestConstants
+            method get
+            proc
+                try
+                begin
+                    Monitor.Enter(lockObject)
+                    begin
+                        if (instance == ^null)
+                            instance = JsonSerializer.Deserialize<TestConstants>(File.ReadAllText("C:\Users\devadm\Desktop\HarmonyCore\Services.Test\TestConstants.Values.json"))
```
Comment (hippiehunter) [inline Services.Test/TestConstants.Properties.dbl:35]: I dont think you want this hard coded here

Comment (theplasmastorm) [reply Services.Test/TestConstants.Properties.dbl]: Changed hard path to env var look up.
See d801f24c78579c0780cd6932d0323b58d16a8d0c

File: Services.Test/TestConstants.Properties.dbl [hunk 2]
```diff
@@ -32,7 +32,17 @@ namespace Services.Test
                     Monitor.Enter(lockObject)
                     begin
                         if (instance == ^null)
-                            instance = JsonSerializer.Deserialize<TestConstants>(File.ReadAllText(Path.Combine(Environment.GetEnvironmentVariable("SOLUTIONDIR"), "Services.Test", "TestConstants.Values.json")))
+                        begin
```
Comment (hippiehunter) [inline Services.Test/TestConstants.Properties.dbl:35]: I don't love throwing exceptions when we should be checking File.Exists first. If file exists and deserialize fails i want that to bubble up to the user not silently fail.

Comment (theplasmastorm) [reply Services.Test/TestConstants.Properties.dbl:35]: Fixed. See 6bf3b98fe5fcf26526643e0b1338d8a169e07e19

File: Services.Test/TestConstants.Properties.dbl [hunk 3]
```diff
@@ -35,9 +35,13 @@ namespace Services.Test
...
                             begin
-                                instance = JsonSerializer.Deserialize<TestConstants>(File.ReadAllText(Path.Combine(Environment.GetEnvironmentVariable("SOLUTIONDIR"), "Services.Test", "TestConstants.Values.json")))
+                                data filePath = Path.Combine(Environment.GetEnvironmentVariable("SOLUTIONDIR"), "Services.Test", "TestConstants.Values.json")
+                                if (File.Exists(filePath)) then
+                                    instance = JsonSerializer.Deserialize<TestConstants>(File.ReadAllText(filePath))
+                                else
+                                    instance = new TestConstants()
                             end
-                            catch (e, @Exception)
+                            catch (e, @JsonException)
```
Comment (hippiehunter) [inline Services.Test/TestConstants.Properties.dbl:44]: I dont think we should be silently failing here, the user should really know about invalid json

Comment (theplasmastorm) [reply Services.Test/TestConstants.Properties.dbl:47]: Fixed. See 57749c739dd187b6a069b54de936a4b2e8f07117

**Pull request merged**